Sources/Vapor/Middleware/TracingMiddleware.swift

```diff
@@ -4,10 +4,21 @@ import Tracing
 ///
 /// See https://opentelemetry.io/docs/specs/semconv/http/http-spans/
 public final class TracingMiddleware: AsyncMiddleware {
-    public init() {}
+    private let setCustomAttributes: @Sendable (inout SpanAttributes, Request) -> Void
+    
+    /// Create a TracingMiddleware
+    /// - Parameter setCustomAttributes: Closure that allows setting custom span attributes for a particular request. A custom span attribute could be extracted from a request
+    /// header, for example. This closure is called during span creation on every request, so should be lightweight.
+    public init(
+        setCustomAttributes: @escaping @Sendable (inout SpanAttributes, Request) -> Void = { _, _ in }
+    ) {
```

Comment:
This will get rid of the API breakage complaint (which is for once technically correct).

```suggestion
    public init() {
        self.setCustomAttributes = { _, _ in }
    }
    
    public init(
        setCustomAttributes: @escaping @Sendable (inout SpanAttributes, Request) -> Void
    ) {
```

Reply:
Oh huh, I guess I thought adding a parameter with a default value was non-breaking. In any case, I've adjusted to your suggestion.

Reply:
Technically it's a break but normally we don't treat it as such in Vapor. It does however keep the API checker quiet